src/hera/env.py

```diff
@@ -55,10 +58,33 @@ class Env:
     value: Optional[Any] = None
     value_from_input: Optional[Union[str, Parameter]] = None
 
+    @staticmethod
+    def _sanitise_param_for_argo(v: str) -> str:
+        """Argo has some strict parameter validation. To satisfy, we replace all ._ with a dash,
+        take only first 32 characters from a-zA-Z-, and append md5 digest of the original string."""
+        # NOTE move this to some general purpose utils?
+        replaced_dashes = v.translate(str.maketrans({e: '-' for e in "_."})) # type: ignore
+        legit_set = string.ascii_letters + '-'
```

Comment:
IIRC, the naming convention followed by Argo follows the one of K8S, and K8S uses the specification of RFC1123. That said, do you happen to know whether the parameter name follows the same convention? If so, perhaps `0-9` can be part of the legit set?

Reply:
good point, you are right, numbers are indeed tolerated (even at the start of the name). So I've just extended the legit set and one of the test cases.